flink-end-to-end-tests/flink-sql-gateway-test/src/test/java/org/apache/flink/table/gateway/SqlGatewayE2ECase.java

```diff
@@ -126,6 +135,339 @@ public void testSqlClientExecuteStatement() throws Exception {
+                                    + "  'default-database' = 'test_db',\n"
+                                    + "  'path' = '%s'\n"
+                                    + ")",
+                            catalogFolder.getAbsolutePath());
+            gatewayRestClient.executeStatementWithResult(createCatalogDDL);
+            gatewayRestClient.executeStatementWithResult("USE CATALOG test_catalog");
+
+            File filesystemSourceFolder = FOLDER.newFolder("filesystem-source");
+
+            gatewayRestClient.executeStatementWithResult(
+                    String.format(
+                            "CREATE TABLE batch_source (\n"
+                                    + "    `timestamp` TIMESTAMP(3),\n"
+                                    + "    `user` VARCHAR,\n"
+                                    + "    `type` VARCHAR\n"
+                                    + " ) with (\n"
+                                    + "    'connector' = 'filesystem',\n"
+                                    + "    'path' = '%s',\n"
+                                    + "    'format' = 'json'\n"
+                                    + " )",
+                            filesystemSourceFolder.getAbsolutePath()));
+
+            gatewayRestClient.executeStatementWithResult(
+                    " CREATE MATERIALIZED TABLE my_materialized_table_in_full_mode\n"
+                            + "                        PARTITIONED BY (ds)\n"
+                            + "                        WITH (\n"
+                            + "                            'partition.fields.ds.date-formatter' = 'yyyy-MM-dd',\n"
+                            + "                            'format' = 'json'\n"
+                            + "                        )\n"
+                            + "                        FRESHNESS = INTERVAL '30' SECOND\n"
+                            + "                        REFRESH_MODE = FULL\n"
+                            + "                        AS SELECT\n"
+                            + "                            ds,\n"
+                            + "                            count(*) as cnt\n"
+                            + "                            FROM (\n"
+                            + "                                SELECT\n"
+                            + "                                    DATE_FORMAT(`timestamp`, 'yyyy-MM-dd') AS ds,\n"
+                            + "                                    user,\n"
+                            + "                                    type\n"
+                            + "                                FROM batch_source\n"
+                            + "                            ) GROUP BY ds");
+
+            long systemTime = System.currentTimeMillis();
+            String todayTimestamp =
+                    formatTimestampMillis(systemTime, "yyyy-MM-dd HH:mm:ss", TimeZone.getDefault());
+            String yesterdayTimestamp =
```

Comment:
What is the function of yesterday?

Reply:
Send data to yerterday for verifying that the refresh operation will not affect historical partitions. I have added relevant verification conditions.